OMCompiler/SimulationRuntime/c/simulation/solver/spatialDistribution.c

```diff
@@ -272,6 +274,23 @@ void storeSpatialDistribution(DATA* data, threadData_t *threadData, unsigned int
     omc_throw_function(threadData);
   }
 
+  /* Get deltaX */
+  deltaX = spatialDistribution->oldPosX - posX;
+  if (deltaX > 0) {
+    realDirection = 1 /* positive */;
+  } else if (deltaX < 0) {
+    realDirection = -1 /* negative */;
+  } else {
+    realDirection = 0 /* standing still */;
+  }
+  deltaX = fabs(deltaX);
```

Comment:
You can even save the `fabs` call while you're at it :smile: 
```suggestion
    realDirection = -1 /* negative */;
    deltaX = -deltaX;
  } else {
    realDirection = 0 /* standing still */;
  }

Reply:
Yes... I could have done that.